app/main/views/services.py

```diff
@@ -506,6 +507,76 @@ def update_section_submission(framework_slug, lot_slug, service_id, section_id,
                                 service_id=service_id))
 
 
+@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/<service_id>/remove/<section_id>/<question_slug>',
+            methods=['GET', 'POST'])
+@login_required
+def remove_subsection(framework_slug, lot_slug, service_id, section_id, question_slug):
+    try:
+        draft = data_api_client.get_draft_service(service_id)['services']
+    except HTTPError as e:
+        abort(e.status_code)
+
+    if not is_service_associated_with_supplier(draft):
+        abort(404)
+
+    content = content_loader.get_manifest(framework_slug, 'edit_submission').filter(draft)
+    question_to_remove = content.get_question_by_slug(question_slug)
+    containing_section = get_containing_section(content, question_to_remove)
+    fields_to_remove = question_to_remove.form_fields
+
+    if request.args.get("confirm") and request.method == "POST":
+        # Remove the section
+        for field in fields_to_remove:
+            draft[field] = None
+
+        try:
+            data_api_client.update_draft_service(
+                service_id,
+                draft,
+                current_user.email_address
+            )
+            flash({'service_name': question_to_remove.label}, 'service_deleted')
+        except HTTPError as e:
+            if e.status_code == 400:
+                # You can't remove the last one
+                flash({
+                    'remove_last_attempted': containing_section.name,
+                    'service_name': question_to_remove.label
+                }, 'error')
+            else:
+                abort(e.status_code)
+
+    else:
```

Comment:
The reason for the checks here is to avoid showing the "Are you sure?" message when clicking "Remove" on the last item. In that case the app needs to decide to show the "You can't remove the last one" without calling the API at all.